src/main/java/org/candlepin/service/CloudRegistrationAdapter.java

```diff
@@ -93,4 +92,29 @@ CloudAccountData setupCloudAccountOrg(String cloudAccountID, String cloudOfferin
         CloudProvider cloudProviderShortName, String ownerKey)
         throws CouldNotAcquireCloudAccountLockException, CouldNotEntitleOrganizationException;
 
+    /**
+     * Create (if one does not already exist) and entitle an owner upstream of Candlepin, for the given
+     * cloud account id and cloud offering id.
+     *
+     * @param cloudAccountID
+     *     Identifier for cloud account that needs access to Red Hat content
+     *
+     * @param cloudProviderShortName
+     *     Shortcut of the cloud provider from which the offering came
+     *
+     * @param cloudOfferingID
+     *     Identifier for a Red Hat cloud offering that the owner needs to be entitled for
+     *
+     * @throws UpstreamOrganizationNotExistException
+     *     Organization does not exist or could not be entitled
+     *
+     * @throws CouldNotEntitleOrganizationException
+     *     Organization could not be entitled
+     *
+     * @return key of the organization
+     */
+    String validateRegisteredOwner(String cloudAccountID, CloudProvider cloudProviderShortName,
```

Comment:
I think `validateCloudAccountOrg` might be a better name for this method as it validates that the organization exists and is entitled based on the offer ID and cloud account. Also, the adapter implementation of this method won't handle the actual organization provisioning and entitling, that will be handled in the `CloudAccountOrgSetupJob`. So, we will want to update the java doc accordingly

Reply:
done